OptimoveSDK/Sources/Classes/PreferenceCenter/OptimovePreferenceCenter.swift

```diff
@@ -0,0 +1,192 @@
+import Foundation
+import OptimoveCore
+
+@available(iOS 13.0, *)
+public class OptimovePreferenceCenter {
+    enum Error: LocalizedError {
+        case alreadyInitialized
+        case notInitialized
+
+        var errorDescription: String? {
+            switch self {
+            case .alreadyInitialized:
+                return "The PreferenceCenterSDK has already been initialized."
+            case .notInitialized:
+                return "Preference center has not been initialized."
+            }
+        }
+    }
+
+    private static var instance: OptimovePreferenceCenter?
+    private var networkClient: NetworkClient?
+    private var storage: OptimoveStorage?
+
+    static var isSdkRunning: Bool {
+        return Optimove.getConfig()?.getPreferenceCenterConfig() != nil
+    }
+
+    public enum ResultType {
+        case success
+        case errorUserNotSet
+        case errorNotConfigured
+        case error
+    }
+
+    public typealias PreferencesGetHandler = (_ result: ResultType, _ preferences: Preferences?) -> Void
+    public typealias PreferencesSetHandler = (_ result: ResultType) -> Void
+
+    public static func getInstance() throws -> OptimovePreferenceCenter {
+        guard let instance = instance else {
+            throw Error.notInitialized
+        }
+        return instance
+    }
+
+    static func initialize(storage: OptimoveStorage, networkClient: NetworkClient) throws {
+        guard instance == nil else {
+            throw Error.alreadyInitialized
+        }
+
+        instance = OptimovePreferenceCenter(storage: storage, networkClient: networkClient)
+    }
+
+    private init(storage: OptimoveStorage, networkClient: NetworkClient) {
+        self.networkClient = networkClient
+        self.storage = storage
+
+        Logger.debug("Preference center SDK was initialized")
+    }
+
+    public func getPreferencesAsync(completion: @escaping PreferencesGetHandler) {
+        guard let config = Optimove.getConfig()?.getPreferenceCenterConfig() else {
+            Logger.error("Preference center credentials are not set")
+            completion(.errorNotConfigured, nil)
+            return
+        }
+
+        guard
+            let customerId = try? storage?.getCustomerID(),
+            let visitorId = try? storage?.getVisitorID(),
+            customerId != visitorId
+        else {
+            Logger.warn("Customer ID is not set")
+            completion(.errorUserNotSet, nil)
+            return
+        }
+
+        Task {
+            do {
+                let request = createGetPreferencesRequest(for: customerId, with: config)
+                let data = try await networkClient?.performAsync(request) ?? Data()
```

Comment:
in which cases ?? Data() happens? what happens next, when decoder tries to get Preferences from it?